### ISSUE TITLE
bug/Unstructured unnecessarily "repairs" then falls back to OCR on extremely large documents

### ISSUE THREAD

**Issue by dhdaines** [Unstructured-IO/unstructured#3815]
TL;DR this is because Unstructured.io applies a ["fix" to pdfminer.six](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/patches/pdfminer.py) which actually [introduces another bug](https://github.com/pdfminer/pdfminer.six/issues/1025).

The fix (which I have tested here) is, until the bug is fixed in pdfminer.six, to *not* apply the patch, and to simply avoid the following versions of pdfminer.six which contain the bug: 20231228 and 20240706

The source of the problem is this change to pdfminer.six (which I think was submitted by unstructured?): https://github.com/pdfminer/pdfminer.six/pull/885

This broke the parsing of long content streams, because due to the complex and not particularly useful buffering done by pdfminer.six's "parser" (actually a lexer), it is frequently the case that keywords are split across a buffer boundary.  In my particular testcase (cannot share, sorry) this results in the "Detected invalid dictionary construct for PDFminer" error reported by Unstructured.io, because a dictionary ends up with an odd number of items as `false` is getting split into `fals` and `e`.

"Repairing" the PDF will not help this problem because there isn't anything wrong with the PDF, it just has some long content streams.  So what happens is that (on my document of 1500+pages) unstructured falls back to OCR, which as you can imagine is ... suboptimal.

**Comment by dhdaines** [comment]
If you *must* patch pdfminer.six, then please apply this fix, which fixes both the original problem you patched and the problem you introduce by patching it: https://github.com/pdfminer/pdfminer.six/pull/1030

**Comment by PhorstenkampFuzzy** [comment]
+1

**Comment by PhorstenkampFuzzy** [comment]
Any update on this?
